_proto/blimp/node/v0/controller.proto

```diff
@@ -16,15 +17,17 @@ service Controller {
   rpc SyncNotifications(stream SyncStatusResponse) returns (stream GetSyncStatusRequest) {}
 }
 
-// In a Tunnel, name, port, and token are used. In an ExposedTunnel, token and
-// namespace are used.
 message TunnelHeader{
-    string name = 1;
-    uint32 port = 2;
-    // For Tunnel, token is a normal OAuth ID Token. For ExposedTunnel, it is
-    // the secret key indexing the service/port combination to be exposed.
-    string token = 3;
-    string namespace = 4;
+  reserved 4;
+  string name = 1;
+  uint32 port = 2;
+  string old_token = 3;
+  blimp.auth.v0.BlimpAuth auth = 5;
+}
+
+message ExposedTunnelHeader{
+  string token = 1;
```

Comment:
Why isn't this blimp.auth.v0.BlimpAuth?

Reply:
Oh I guess because we don't support self hosted link sharing? If that's the reason can you put an XXX comment here?

Reply:
This "token" is not a blimp token. It's the index into the expose annotation on the namespace. Replacing with BlimpAuth wouldn't make sense, just like we don't replace the "token" field in the KubeCredentials message (because it is a different kind of token).

Reply:
This "token" is not a blimp token. It's the generated secret token corresponding to the exposed service/port. Replacing with BlimpAuth wouldn't make sense, just like we don't replace the "token" field in the KubeCredentials message (because it is a different kind of token).

Reply:
Ah got it